server/src-lib/Hasura/RQL/Types/Subscribe.hs

```diff
@@ -122,11 +122,11 @@ instance FromJSON CreateEventTriggerQuery where
     webhook   <- o .: "webhook"
     headers   <- o .:? "headers"
     replace   <- o .:? "replace" .!= False
-    let regex = mkRegex "^\\w+$"
+    let regex = mkRegex "\\W"
```

Comment:
@nathanstitt Can you please change this to the following condition:
`^[A-Za-z]+[A-Za-z0-9_\\-]*$`

Not only is this more safe (vanilla regex) but it also allows for `-` (hyphen). Must be merged after https://github.com/hasura/graphql-engine/pull/1012/ .

Reply:
Seems like we will need to change the whole regex package. Turns out we need a native Haskell package like regex-tdfa : https://wiki.haskell.org/Regex_Posix

```
If you want a bug-free and/or portable POSIX extended regular expression library to use from Haskell, then regex-posix will not help you. You should use the regex-tdfa package instead.
```

@nathanstitt If you are up for it, you may try changing the package from `regex-compat` to `regex-tdfa` . Otherwise I'll queue this up in my list.